client/overview/task-list/dispute-tasks.ts

```diff
@@ -0,0 +1,221 @@
+import wcpayTracks from 'wcpay/tracks';
+
+/**
+ * Returns true if the dispute is due within the specified number of days.
+ *
+ * @param {CachedDispute} dispute - The dispute to check.
+ * @param {number} [days] - The number of days to check. Defaults to 1.
+ *
+ * @return {boolean} True if the dispute is due within the specified number of days.
+ */
+const isDueWithin = ( dispute: CachedDispute, days = 1 ) => {
+	const now = moment();
+	const dueBy = moment( dispute.due_by );
+	return (
+		dueBy.diff( now, 'hours' ) > 0 &&
+		dueBy.diff( now, 'hours' ) <= 24 * days
+	);
+};
+
+/**
+ * Returns an array of disputes that are due within the specified number of days.
+ *
+ * @param {CachedDispute[]} activeDisputes - The active disputes to filter.
+ * @param {number} days - The number of days to check.
+ *
+ * @return {CachedDispute[]} The disputes that are due within the specified number of days.
+ */
+export const getDisputesDueWithinDays = (
+	activeDisputes: CachedDispute[],
+	days: number
+): CachedDispute[] =>
+	activeDisputes.filter( ( dispute ) => isDueWithin( dispute, days ) );
+
+export const getDisputeResolutionTask = (
+	/**
+	 * Active disputes (awaiting a response) to generate the notice string for.
+	 */
+	activeDisputes: CachedDispute[]
+): TaskItemProps | null => {
+	const disputeCount = activeDisputes.length;
+
+	if ( disputeCount === 0 ) {
+		return null;
+	}
+
+	const handleClick = () => {
+		wcpayTracks.recordEvent( 'wcpay_overview_task', {
+			task: 'dispute-resolution-task',
+		} );
+		const history = getHistory();
+		if ( disputeCount === 1 ) {
+			// Redirect to the dispute details page if there is only one dispute.
+			const disputeId = activeDisputes[ 0 ].dispute_id;
+			history.push(
+				getAdminUrl( {
+					page: 'wc-admin',
+					path: '/payments/disputes/details',
+					id: disputeId,
+				} )
+			);
+		} else {
+			history.push(
+				getAdminUrl( {
+					page: 'wc-admin',
+					path: '/payments/disputes',
+					filter: 'awaiting_response',
+				} )
+			);
+		}
+	};
+
+	// Create a new array and sort by `due_by` ascending.
+	activeDisputes = [ ...activeDisputes ].sort( ( a, b ) =>
+		moment( a.due_by ).diff( moment( b.due_by ) )
+	);
+
+	const numDisputesDueWithin24h = getDisputesDueWithinDays(
+		activeDisputes,
+		1
+	).length;
+
+	// Create a unique key for each combination of dispute IDs
+	// to ensure the task is rendered if a previous task was dismissed.
+	const disputeTaskKey = `dispute-resolution-task-${ activeDisputes
+		.map( ( dispute ) => dispute.dispute_id )
+		.join( '-' ) }`;
+
+	const disputeTask: TaskItemProps = {
+		key: disputeTaskKey,
+		title: '', // Title text defined below.
+		content: '', // Subtitle text defined below.
+		level: 1,
+		completed: false,
+		expanded: true,
+		isDismissable: false,
+		showActionButton: true,
+		actionLabel: __( 'Respond now', 'woocommerce-payments' ),
+		action: handleClick,
+		onClick: () => {
+			// Only handle clicks on the action button.
```

Comment:
Do you think we need to handle CSS on hover? Currently, the background becomes grey as if the whole task is clickable. IMO it's OK to leave it as is though.

Reply:
It is _sort of_ consistent with the other task list items:
* The collapsible tasks toggles collapsing on click
* The completed task does nothing on click, is still grey on hover
* The new dispute task does nothing on click, is still grey on hover

https://github.com/Automattic/woocommerce-payments/assets/3147296/7cb46993-74d7-41af-9005-294a379c1d5b

Reply:
Good point to be consistent with other tasks.